### DISCUSSION TITLE
Error: SEC_ERROR_REUSED_ISSUER_AND_SERIAL

### DISCUSSION THREAD

**Discussion by grzegorzSowa3** [synchronizing/mitm#23] [Q&A] [answered]
Hi, I am getting error: SEC_ERROR_REUSED_ISSUER_AND_SERIAL  (Firefox 107.0 , Windows 10) 
`An error occurred during a connection to www.facebook.com. You are attempting to import a cert with the same issuer/serial as an existing cert, but that is not the same cert.

Error code: SEC_ERROR_REUSED_ISSUER_AND_SERIAL`
Do you have any idea how to fix that? :)

**Comment by synchronizing** [comment]
Hey. A question: are you installing `mitm`'s generated certificates?

**Reply by grzegorzSowa3** [reply]
Yes, I took root CA .pem file and added it to trusted CAs in the browser.
The issue might be in `serial_number: int = random.randint(0, 2 ** 64 - 1)` as a default function argument - I verified it generates random number just once and then it is always the same (I'm using Python 3.10.4).
However, when I changed that I must have broke something else, so now I am getting SEC_ERROR_BAD_SIGNATURE

**Reply by synchronizing** [reply]
Ah, good catch. I'll push a fix up.

**Comment by synchronizing** [comment, marked as answer]
Fixed added in #24. Version [1.4.2](https://pypi.org/project/mitm/) is now live on PyPi.

```
pip install --upgrade mitm
```

Thank you for the bug report.

**Reply by synchronizing** [reply]
Marking as answer for the meantime. Lmk if this didn't fix your issue and I'll reopen the thread.

**Comment by grzegorzSowa3** [comment]
Cool, thanks for the quick answer. Very nice work you do btw, keep it up! :)

**Reply by synchronizing** [reply]
Ofc, and thanks. 😄
